### PR TITLE
DM-653 add permissions check for terms accepted

### PR DESCRIPTION
**Pull request by sutherlanda**
This PR adds a permissions check on certain actions by Vendor users to ensure they have accepted the terms and conditions prior to:

* Creating an organization
* Creating/submitting a CWU proposal
* Creating/submitting a SWU proposal

I didn't add the check to editing/deleting a proposal as we already check for authorship on the proposal, which would imply they had permission to create it.

### CODE REVIEW

File: src/back-end/lib/permissions.ts [hunk 1]
```diff
@@ -294,12 +298,12 @@ export async function readSWUProposalScore(connection: Connection, session: Sess
...
+export async function createSWUProposal(connection: Connection, session: Session): Promise<boolean> {
+  return isVendor(session) && await hasAcceptedTerms(connection, session);
 }
 
 export async function submitSWUProposal(connection: Connection, session: Session, organization: Organization): Promise<boolean> {
-  return isVendor(session) && !!session && await isUserOwnerOfOrg(connection, session.user, organization.id) && doesOrganizationMeetSWUQualification(organization);
+  return !!session && isVendor(session) && await hasAcceptedTerms(connection, session) && await isUserOwnerOfOrg(connection, session.user, organization.id) && doesOrganizationMeetSWUQualification(organization);
 }
 
 export async function editSWUProposal(connection: Connection, session: Session, proposalId: string, opportunity: SWUOpportunity): Promise<boolean> {
```
Comment (dhruvio) [inline src/back-end/lib/permissions.ts:309]: I think we'll need to do the check here. If a vendor is editing a submitted proposal before the proposal deadline, they are technically "resubmitting" a proposal. Applies to both CWU and SWU.

**Review by dhruvio — CHANGES_REQUESTED** [review]
Looks good, one change requested, thanks!

File: src/back-end/lib/permissions.ts (hunk 1 above)
Comment (sutherlanda) [reply src/back-end/lib/permissions.ts:309]: Added to editing of proposals as well.

**Review by dhruvio — APPROVED** [review]

**Comment by dhruvio** [comment]
Looks good to me

**Pull request merged**